doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,293 @@
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * the developer would rebuild their bundle image passing in their scorecard configuration file and their test pod manifests
+ * the Basic and OLM test manifests would be added automatically to the bundle image as part of the bundle image build process.  This allows users to run the Basic and OLM tests along with their custom tests.
+ * lastly, the developer would run the scorecard passing in their operator bundle image they just built along with any other scorecard flags (e.g. selector).
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are Pod manifests in YAML that run a custom test image.  End users would add their test pod manifests and scorecard configuration file into their operator bundle image using a CLI command developed as part of this proposal.
+
+### Scorecard Configuration
+
+Note that it would also be possible to add a resource limit to the scorecard configuration as shown in the example below.  Those resource limits would be applied to custom tests that are executed allowing for some control as to how much resources are available for the test execution.
+
+Custom test Pod manifests would be accessible from a local disk location to scorecard.  Scorecard would obtain the custom test manifests as part of its bundle image processing since the bundle image is passed into scorecard as a configuration flag.  
+
+Scorecard would assume that the scorecard configuration file and test
+manifests would be located in the bundle image at the following location:
+```
+/bundle/scorecard/.osdk-scorecard.yaml
+/bundle/scorecard/tests
+```
+
+Test Pod manifests would be configured from within the scorecard configuration file.  The following is a scorecard configuration file example using this scheme:
+
+```
+scorecard:
+  # Setting a global scorecard option
+  output: text
+  tests:
+    init-timeout: 60
+    resources:
+      limits:
+        cpu: "1"
+        memory: "200Mi"
+      requests:
+        cpu: 750m
+        memory: "100Mi"
```

Comment:
Since the test definitions are pod manifests that support embedding the resources directly, do we need this to be defined and managed by scorecard?

Reply:
the only reason to define them at the scorecard configuration level itself is if you want to enforce those particular limits across all tests, but problems might arise with this plan if you had tests that had to have different resource limits in order to run, so this might be adding extra complexity than necessary for an initial implementation.

Reply:
My vote would be to rely on pods to define their own limits. If it becomes apparent that global resource definitions are necessary, we can add it later.

Reply:
I removed the resources concept in this latest commit.